### ISSUE TITLE
Not working propperly under CPU/RAM usage - Confirmed connection 

### ISSUE THREAD

**Issue by mironkraft** [sebaxzero/Juridia#1]
Hello there Sebax

Maybe we are wrong about the usage of this app, but, afaik we seen, should run under CPU/RAM, is that right?

I was yesterday setting up and making some test, all kind of failed, some screenshots:

![image](https://github.com/sebaxzero/Juridia/assets/42625283/8e520c60-adab-4428-b55f-d9e4b11bf5cd)

![image](https://github.com/sebaxzero/Juridia/assets/42625283/35b7265d-743b-4337-9ad4-98620394e22f)

Next image, shows that he read the PDF I uploaded:

![image](https://github.com/sebaxzero/Juridia/assets/42625283/97a3648e-7b65-4a18-92ae-3173174f0fda)

![image](https://github.com/sebaxzero/Juridia/assets/42625283/a169fcb7-43b3-4a4e-87ff-d7635f5765a2)

Are we doing it wrong by using it under CPU/RAM? It is supposed to work only with GPU?

Thanks, kind regards!

**Comment by mironkraft** [comment]
Late tries goes like this:

![image](https://github.com/sebaxzero/Juridia/assets/42625283/76cebd69-f1a7-437c-b32f-b20014ae36c6)

Also, other error that I get:

UniqueConstraintError: Collection 6d029a2e-76e8-40ad-9ad0-34b35a807bf5 already exists

`File "C:\Users\ASUS\Desktop\IA\oobabooga_windows\installer_files\env\lib\site-packages\streamlit\runtime\scriptrunner\script_runner.py", line 552, in _run_script
    exec(code, module.__dict__)
File "C:\Users\ASUS\Desktop\IA\Juridia-main\st_interface_en.py", line 188, in <module>
    main()
File "C:\Users\ASUS\Desktop\IA\Juridia-main\st_interface_en.py", line 70, in main
    st.session_state.db = VectorStore.get(name=name, chunk_size=chunk_size, chunk_overlap=chunk_overlap, model_name=repo_id, device=device)
File "C:\Users\ASUS\Desktop\IA\Juridia-main\src\main.py", line 142, in get
    vectorstore = VectorStore.create(persist_directory=index_dir, splitted_docs=splitted_docs, embeddings=embeddings, name=name)
File "C:\Users\ASUS\Desktop\IA\Juridia-main\src\main.py", line 79, in create
    return Chroma.from_documents(documents=splitted_docs, embedding=embeddings, persist_directory=persist_directory, collection_name='name')
File "C:\Users\ASUS\Desktop\IA\oobabooga_windows\installer_files\env\lib\site-packages\langchain\vectorstores\chroma.py", line 578, in from_documents
    return cls.from_texts(
File "C:\Users\ASUS\Desktop\IA\oobabooga_windows\installer_files\env\lib\site-packages\langchain\vectorstores\chroma.py", line 533, in from_texts
    chroma_collection = cls(
File "C:\Users\ASUS\Desktop\IA\oobabooga_windows\installer_files\env\lib\site-packages\langchain\vectorstores\chroma.py", line 115, in __init__
    self._collection = self._client.get_or_create_collection(
File "C:\Users\ASUS\Desktop\IA\oobabooga_windows\installer_files\env\lib\site-packages\chromadb\api\segment.py", line 153, in get_or_create_collection
    return self.create_collection(
File "C:\Users\ASUS\Desktop\IA\oobabooga_windows\installer_files\env\lib\site-packages\chromadb\api\segment.py", line 134, in create_collection
    self._sysdb.create_collection(coll)
File "C:\Users\ASUS\Desktop\IA\oobabooga_windows\installer_files\env\lib\site-packages\chromadb\db\mixins\sysdb.py", line 96, in create_collection
    raise UniqueConstraintError(`

**Comment by mironkraft** [comment]
Second Test is kinda working but not ok, it has processed and answered, but cannot ask more:

![image](https://github.com/sebaxzero/Juridia/assets/42625283/33e82edf-a7aa-41ca-9067-4af9abd8d31f)

![image](https://github.com/sebaxzero/Juridia/assets/42625283/a3192c86-4dcc-459d-a11a-3a45ada4bec9)
![image](https://github.com/sebaxzero/Juridia/assets/42625283/7bc1f539-85e7-4b3d-beb9-06ded8e0ca35)

**Comment by mironkraft** [comment]
as you can see, it works on the CMD, but it gets blocked to interact with on the interface:
![image](https://github.com/sebaxzero/Juridia/assets/42625283/eb59b3e5-a907-4799-8773-4ae7f4661021)

![image](https://github.com/sebaxzero/Juridia/assets/42625283/7f50e2bb-49a9-469a-b382-3f747bccbd1c)

**Comment by sebaxzero** [comment]
1. for UniqueConstraintError: Collection 6d029a2e-76e8-40ad-9ad0-34b35a807bf5 already exists.
theres is an issue with how langchain saves multiples vectorstore, im trying to solve it but for now delete "sessions" folder where you cloned the repo, this will delete the process document and you will need to process again.

2. for no response from the llm backend (oobabooga error 404), this is an error from the textgen itself, maybe you didnt enable the api or used a wrong url, if you are using cpu-only i might suggest to use llama.cpp directly instead of oobabooga

**Comment by mbaggnx** [comment]
> 2\. llama.cpp

hello there, we are still fighting to get this work jejeje, when you mean "use llama.cpp" you mean the option inside of oobabooga right?
If yes, It does not loads correctly:

![image](https://github.com/sebaxzero/Juridia/assets/119416732/3382f387-560e-48bb-8759-e6a080ed05fe)

**Comment by sebaxzero** [comment]
> > 2. llama.cpp
> 
> hello there, we are still fighting to get this work jejeje, when you mean "use llama.cpp" you mean the option inside of oobabooga right? If yes, It does not loads correctly:

no, i meant to install it as a python package in the project, without using oobabooga.
you can use oobabooga with llama.cpp but it never work for me so i cant help you much with that.

you can create a python virtual enviroment and use it from there as shown in [#installation](https://github.com/sebaxzero/Juridia#installation)

to install:
1. open cmd, 2. clone repo, 3. navigate, 4. create venv, 5. activate venv, 6. install requirements, 7. install llama.cpp. 

to use:
1. activate venv, 2. streamlit run command.

llama.cpp is a library to run ggml models in cpu-ram, oobabooga allows you to use it but only with compatible ggml models, the error you are getting might be a wrong model format, the model you are trying to load i think is a sentence transformer model, not an llm, for example you can use: [TheBloke/vicuna-13b-v1.3.0-GGML](https://huggingface.co/TheBloke/vicuna-13b-v1.3.0-GGML).
